mounts.go

```diff
@@ -0,0 +1,169 @@
+package testcontainers
+
+import (
+	"github.com/docker/docker/api/types/mount"
```

Comment:
As I understood the already existing code the things related to particular container backend (in this case - docker) are kept in the `docker.go` so that the testcontainers implementation is decoupled from it (although there is right now only one container backend provider).

Reply:
I see. To be honest after [this comment](https://github.com/testcontainers/testcontainers-go/issues/336#issuecomment-891667819) I assumed it would be okay to rely on Docker API types but I refactored my draft to be Docker API independent, moved all the Docker related code to `mounts_docker.go` and split the mount sources into generic, container provider agnostic variants and added Docker specific ones in case one would like to be able to provide e.g. volume options.

I hope this matches better the current code structure?